kvbc/include/categorization/blocks.h

```diff
@@ -0,0 +1,159 @@
+#include "updates.h"
+#include <utility>
+#include <block_digest.h>
+#include "details.h"
+#include "merkle_tree_serialization.h"
+#include "merkle_tree_key_manipulator.h"
+#include "sliver.hpp"
+
+namespace concord::kvbc::categorization {
+
+// Block is composed out of:
+// - BlockID
+// - Parent digest
+// - Categories' updates_info where update info is essentially a list of keys, metadata on the keys and a hash of the
+// category state.
+struct Block {
+  static const std::string CATEGORY_ID;
+
+  Block() = default;
+  Block(const BlockId block_id) { data.block_id = block_id; }
+
+  void add(const std::string& category_id, MerkleUpdatesInfo&& updates_info) {
+    data.categories_updates_info.emplace(std::make_pair(category_id, std::move(updates_info)));
+  }
+
+  void add(const std::string& category_id, KeyValueUpdatesInfo&& updates_info) {
+    data.categories_updates_info.emplace(std::make_pair(category_id, std::move(updates_info)));
+  }
+
+  void add(SharedKeyValueUpdatesInfo&& updates_info) { data.shared_updates_info = std::move(updates_info); }
+
+  void setParentHash(const BlockDigest& parent_hash) {
+    std::copy(parent_hash.begin(), parent_hash.end(), data.parent_digest.begin());
+  }
+  inline BlockId ID() const { return data.block_id; }
+
+  static const serialization::buffer& serialize(const Block& block) {
```

Comment:
Btw, can that be a member (non-static) method?

Reply:
yeah, I made it on purpose, seems nice to have as a utility function under the Block name.
what do you think?

Reply:
I was thinking you need an object to call it and maybe you can make it a `const` method. No big deal.